### DISCUSSION TITLE
Scene Execution in Homekit

### DISCUSSION THREAD

**Discussion by MattGeale** [aarons22/homebridge-bond#194] [Q&A]
Don't know if its just my setup on Homekit but homekit is not executing the "raise" command on my blinds that I've setup with the pro. I know the scene is working because it turns off my TV, but it does not raise the blinds when scene is turned off. When the blinds are up, the scene executes perfectly...drops the blinds I've nominated as well as turn on the TV, but when turning off the scene, the blinds remain down, but everything else turns off/reverts to the original state (ie TV turns off)... Any help would be greatly appreicated.
Cheers!
![image](https://user-images.githubusercontent.com/88370061/147843531-f02292fa-18ab-4a1b-b391-02eb32e88c2c.png)

**Comment by chrismerck** [comment]
Few questions:

1. Presumably all the individual shades work OK via HomeKit?
2. After executing the problem scene, does the problem shade report in the desired state in the Bond Home app? That is, does Bond Bridge think that it already adjusted the shade? Or did it fail to get the command from HomeBridge?
3. Do you use any other integration like Google Home or Alexa? If so, what happens if you ask to raise/lower all the shades? Does it work?

What I'm thinking is, the shades might not be appreciating rapid-fire signals and may need additional padding between transmissions. This is something on the Bond side that we have to tweak from time to time for certain shades. Depends on the shade technology (Somfy RTS, Rollease, etc.). -- Or it could be a HomeKit/HomeBridge/configuration. Above questions should clarify.

**Comment by MattGeale** [comment]
Hey @chrismerck,

Answering your questions:
1. yep when triggering the above scene when the blinds are up, the corresponding blinds all respond to the drop command. Its the command to send them up when the scene turns off that they're not responding. To combat that, and as a stop-gap measure while this thread is answered,I created another scene that I trigger to raise the above blinds. Works, but in a perfect world, the above scene to drop and raise the blinds would be my preference.
2. When I trigger the scene to turn off, the Bond App is not showing the required state (ie raise the blinds). My guess is the command is failing to be sent via homebridge. 
3. No - I havent configured Home or Alexa as my preferred home assistant is Siri (granted, I'm triggering the scenes from my phone via Siri and/or the home app)

These are my shade motors - [Sonesse 30 WireFree RTS (Li-ion)](https://www.somfysystems.com/en-us/products/1003128/sonesse-30wirefree-rts-li-ion)

Cheers!

**Reply by chrismerck** [reply]
> When I trigger the scene to turn off, the Bond App is not showing the required state (ie raise the blinds). My guess is the command is failing to be sent via homebridge.

Ok, that is the key piece of information, strongly suggesting that the problem here is somewhere in the Homekit-HomeBridge or HomeBridge-BondBridge connections. Some logs from HomeBridge side or Bond Bridge side could be helpful in further debugging. You could dig into the Homebridge logs and see if all the shades you want to control have associated requests to Bond, and what the status code of those requests are.

**Reply by MattGeale** [reply]
@chrismerck apologies for the delay. attached are the homebridge logs while running in debug mode. You'll see in the latter half of the screen, around about 06:37 and 06:38 theres a minute gap. I manually  created a scene there to open them as they wern't opening hence the 'successful' message received/sent at the very end. let me know your thoughts and what I can do to assist. Cheers!
[new 3.txt](https://github.com/aarons22/homebridge-bond/files/7817667/new.3.txt)

**Comment by MattGeale** [comment]
@chrismerck anything you can see in the logs that explain as to why the blinds aren't opening when the scene is turned off? Thanks!

**Reply by chrismerck** [reply]
You would have to go line-by-line in the logs and think about what it means in your situation.

**Reply by MattGeale** [reply]
I don't know much about logs nor can I read them, but from what I can see, the homebridge isn't either recording the scene change (don't know if homebridge even recognises scenes) or the homebridge - bond connection isn't working (which is what you came to the assumption on already based on how they're not getting a raise command from the bond.

**Comment by MattGeale** [comment]
@chrismerck I still haven't had much luck in trying to narrowdown whats occuring on the Homebridge side - @aarons22 any ideas or do you need some further info? Apologies for the mentions, but you two are the most knowledgeable I've found. 
Cheers
